src/palace/manager/sqlalchemy/model/licensing.py

```diff
@@ -1669,7 +1669,10 @@ def set(
         if not dirty and update_available and available is not None:
             lpdm.available = available
 
-        if not lpdm.rights_status or rights_status.uri != RightsStatus.UNKNOWN:
+        if not lpdm.rights_status or (
+            lpdm.rights_status != rights_status
```

Comment:
Without this check for non-equality, this test was returning true in almost every case leading to returning a dirty value for the delivery mechanism in every case which meant that 60 odd license pools were being updated unnecessarily for every book update.  So that means that the set_open_access_status was being called 3600  the georgia collections (60 license pools * 60 import tasks )  unnecessarily for every single book.  If were updating 28000 books in each of 12  collections (ie complete reimport for those 12 only) we'd see 1.2 billion unnecessary calls.

Reply:
When you consider this method is also called by the reaper, we're taking 28000*60*3600 = 6 billion unnecessary calls every run of the reaper.